app/controllers/notes_controller.rb

```diff
@@ -40,6 +44,12 @@ def destroy
 
   private
 
+  def search_key
+    if params[:search_key].present?
+      params[:search_key]
+    end
+  end
```

Comment:
The presence check is redundant here since the lookup will already return a `nil` if the param is missing.

Reply:
This is to check for blank and spaces.  So, if the value of search_key is either blank or space then we can avoid a search for it in DB.